website/content/en/docs/advanced-topics/heterogeneous.md

```diff
@@ -0,0 +1,92 @@
+---
+title: Heterogeneous Cluster Support
+linkTitle: Heterogeneous Cluster Support
+weight: 80
+---
+## Overview
+A heterogeneous cluster is a cluster in which the nodes differ from each other in a way that makes their containers incompatible with each other. The most common incompatibility that applies is CPU architecture. The rest of this document assumes that the user is developing an application that needs to be able to deploy cleanly in a cluster with a mix of node archictectures. Because an operator deployed to a heterogeneous cluster may not support all of the node architectures in the cluster, it is important to set up affinity safeguards that prevent scheduling to incompatible architectures. These scheduling implications affect both the operator as well as its operands.
+
+### Setting the Supported Architectures
+
+Kubernetes provides a mechanism called `nodeAffinity` which can be used to limit the possible node targets where a pod can be scheduled. Here is an example of how to update a `PodSpec` or `PodTemplateSpec` to prevent the scheduling of pods on nodes of incompatible architectures. Here we compare the `kubernetes.io/arch` key to ensure that the value falls within on of the supported architectures. This assumes that the pod in question is referencing a manifest list which points to container images for each of the architectures listed. It is important to remember that `nodeAffinity` must be set anywhere a container reference is defined, including `Pod`, `Deployment`, `DaemonSet`, `StatefulSet`, or any other object that defines a `PodSpec` or `PodTemplateSpec`.
+
+```yaml
+affinity:
+  nodeAffinity:
+    requiredDuringSchedulingIgnoredDuringExecution:
```

Comment:
Why the requiredDuringSchedulingIgnoredDuringExecution is better than preferredDuringSchedulingIgnoredDuringExecution?

Reply:
Added a sentence after this block explicitly answering that question. :)